### PR TITLE
Bulk upload timesheet to Xero

### PR DESCRIPTION
**Pull request by khoido**

### CODE REVIEW

**Comment by thukim** [comment]
could you pls add unit tests for bulk_ts and ts sync classes?

File: lib/eh_xero/sync_service.rb [hunk 1]
```diff
@@ -67,6 +67,16 @@ def pull(context, type: )
...
       end
 
+      def upload_bulk_timesheet(context, timesheets, options = {})
+        _, gateway = prepare(context)
+        Sync::BulkTimesheet.upload(gateway, timesheets, options)
+      end
+
+      def upload_timesheet(context, timesheet, options = {})
+        context, gateway = prepare(context)
+        Sync::Timesheet.upload(gateway, context,  timesheet, options)
```
Comment (khoido) [inline lib/eh_xero/sync_service.rb]: yep, the `context` is only for logging purpose

File: lib/eh_xero/xeroizer_customs.rb [hunk 2]
```diff
@@ -22,5 +23,9 @@
 require "eh_xero/xeroizer_customs/models/payslips/tax_line"
 require "eh_xero/xeroizer_customs/models/payslip"
 require "eh_xero/xeroizer_customs/models/pay_run"
+require "eh_xero/xeroizer_customs/models/timesheet"
+require "eh_xero/xeroizer_customs/models/timesheet_line"
+require "eh_xero/xeroizer_customs/models/number_of_unit"
```
Comment (khoido) [inline lib/eh_xero/xeroizer_customs.rb]: yea, they are the dependencies of `timesheet` model

File: lib/eh_xero/sync/timesheet.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+module EhXero
+  module Sync
+    class Timesheet
+      extend Base
+
+      class << self
+        def upload(gateway, context, eh_timesheet,  options = {})
+          xero_timesheet = gateway.Timesheet.build(eh_timesheet)
+          xero_timesheet.save
```
Comment (khoido) [inline lib/eh_xero/sync/timesheet.rb]: To be honest, I think we should not use http to request big data from EH. It will slow down the performance a lot.

**Pull request merged**